core/trino-main/src/main/java/io/trino/operator/OutputSpoolingController.java

```diff
@@ -113,11 +140,15 @@ public void recordEncoded(long encodedSize)
         spooledEncodedBytes += encodedSize;
     }
 
-    public void recordInlined(int positionCount, long sizeInBytes)
+    public void recordInlined(AtomicLong totalInlinedPositions, AtomicLong totalInlinedSize, int positionCount, long sizeInBytes)
     {
         inlinedPositions += positionCount;
         inlinedPages++;
         inlinedRawBytes += sizeInBytes;
+
+        // Increase per-pipeline counters
+        totalInlinedPositions.getAndAdd(positionCount);
+        totalInlinedSize.getAndAdd(sizeInBytes);
```

Comment:
To me it looks like we are mixing two kinda separate responsibilities in single class.
* Governing what should be the mode of results returining (this should be done at pipeline level)
* Collecting per operator spooling related stats.

Lets split this to two separate classes.